fv/infrastructure/infra_k8s.go

```diff
@@ -338,6 +411,31 @@ func (kds *K8sDatastoreInfra) AddNode(felix *Felix, idx int, needBGP bool) {
 	}
 }
 
+func (kds *K8sDatastoreInfra) ensureNamespace(name string) {
+	if isSystemNamespace(name) {
+		return
+	}
+
+	ns := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+	_, err := kds.K8sClient.CoreV1().Namespaces().Create(ns)
+	if err != nil {
+		panic(err)
```

Comment:
Is this function only called when we know that the namespace does not yet exist?

If it is possible for the namespace already to exist, I guess we would need to allow that error here.

Reply:
Yes.